### ISSUE TITLE
Separate type and details when adding appointment

### ISSUE THREAD

**Issue by christofferengstrom** [agensdev/WomansHealth#3]
![add new](https://cloud.githubusercontent.com/assets/5484536/24695115/ca25f772-19e4-11e7-92f7-6cd5bc69344e.png)

[Link to project in inVision](https://projects.invisionapp.com/d/main#/projects/prototypes/10710845)

- [x] When pressing "+" the user gets the list of suggested types of appointments.

- [x] The view should have a title and subtitle explaining what the user sees.

- [x] "Save"-button should have a disabled state when required fields have not yet been filled out.

**Comment by blashca** [comment]
While working on this issue, I thought it would be good to include last date of the medical test, as it was in previous app version (screenshot below). I think it makes it easer to keep track of all medical tests in time. Also, for example pregnant women, do some test repeatedly every month, that's why I've add possibility of setting next date based on last date of the test. What do you think?
![screenshot_20170413-170321](https://cloud.githubusercontent.com/assets/9930565/25011045/4641ccea-206c-11e7-9a6c-69b9dd598c26.png)

**Comment by hfossli** [comment]
Yep, the design is missing that aspect. Any thoughts @christofferengstrom? 

FYI: We're on holidays until Tuesday so I don't have access to android studio etc

**Comment by hfossli** [comment]
@christofferengstrom is out of office today so he will be able to answer this on thursday I beleive

**Comment by christofferengstrom** [comment]
That's true. My though was that if you have previous appointments the show up automatically, instead of having to fill out manually. I know the flow is limited to first time experience..

When it comes to next date I agree that it's usefull! I just think the current UX is a bit confusing when seeing it for the first time..
- Next date
- Off
- from last date

I think a better solution would be:
If you have a previous appointment, and you tap "select date" --> You get a picker with preferred choices (1 month from, 6 month, 1 year -- from 3 jan 2017)
And in the bottom you can choose "custom", where you get the regular date picker.

But that's out of scope! 😁 
So I agree, better to keep the one you had than removing this feature.

**Comment by hfossli** [comment]
Yes, I too find it a bit confusing that this screen is covering two use cases (tracking for history and setting up reminders). The reminders is the core feature of the app if I understand it correctly. 

I think it might be better to separate these two.

**Comment by hfossli** [comment]
Not sure where to go from here. What are your thoughts Blanka?

**Comment by blashca** [comment]
I think you're right that it's confusing to cover tracking history and reminders in one place. So maybe a good way of solving this would be to add some kind of history records button to track old appointments (like in weight/bmi). Since it's out of scope for now, how about I'll make the changes according to @christofferengstrom's initial suggestions/mockups and leave the history as a to do for later?

**Comment by hfossli** [comment]
I had a chat about this with @christofferengstrom and we think maybe it will be best to just leave this screen (appointment details) as it currently is implemented. If we were to make big changes like this we should have a workshop about it first. 

I think @christofferengstrom has some final QA feedback on the template-list before we can close this. :)

**Comment by blashca** [comment]
In the meantime, I've improved colors in adding appointment activity, to match new app design. And also added enabling "save" button after all needed input is filled.

**Comment by blashca** [comment]
I've opened a separate PR (#7) for the above.

**Comment by blashca** [comment]
Since PR #7 is merged I think we can close that as well.
